contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# This script only gets executed on systemd systems, see mount-zfs.sh for non-systemd systems
+
+# There is a race between the zpool import and the pre-mount hooks, so we
+#   we wait for a pool to be imported
+while true; do
+    if $(zpool list -H | grep -q -v '^$'); then
+        break
+    fi
+    if $(systemctl is-failed zfs-import-scan.service) == 'failed' ]]; then
+      exit 1
+    fi
+    if [[ $(systemctl is-failed zfs-import-cache.service) == 'failed' ]]; then
+      exit 1
+    fi
+    sleep 0.10s
+done
+
+# import the libs now that we know the pool imported
+[ -f /lib/dracut-lib.sh ] && dracutlib=/lib/dracut-lib.sh
+[ -f /usr/lib/dracut/modules.d/99base/dracut-lib.sh ] && dracutlib=/usr/lib/dracut/modules.d/99base/dracut-lib.sh
+. "$dracutlib"
+
+[ -z "$root" ] && root=$(getarg root=)
+# If root is not ZFS= or zfs: or rootfstype is not zfs
+# then we are not supposed to handle it.
+[ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
+if [ "${root}" = "zfs:AUTO" ] ; then
+    root=$(zpool list -H -o bootfs)
+else
+    root="${root##zfs:}"
+    root="${root##ZFS=}"
+fi
+
+# if pool encryption is active and the zfs command understands '-o encryption'
+if [[ $(zpool list -H -o feature@encryption) == 'active' ]]; then
```

Comment:
See the previous comment about specifying the pool name.

Reply:
should be able to derive it from `${root}`